src/diploma.c

```diff
@@ -0,0 +1,254 @@
+#include "gba/macro.h"
+#include "global.h"
+#include "gpu_regs.h"
+#include "window.h"
+#include "main.h"
+#include "scanline_effect.h"
+#include "task.h"
+#include "unk_text_util.h"
+#include "gpu_regs.h"
+#include "text.h"
+#include "bg.h"
+#include "sound.h"
+#include "malloc.h"
+#include "sprite.h"
+#include "palette.h"
+#include "new_menu_helpers.h"
+#include "menu.h"
+#include "diploma.h"
+
+void sub_80568FC(void);
+u16 sub_8088F84(void);
+
+void DiplomaBgInit(void);
+void DiplomaPrintText(void);
+u8 DiplomaLoadGfx(void);
+void DiplomaVblankHandler(void);
+
+void CB2_DiplomaInit(void);
+
+void Task_WaitForExit(u8);
+void Task_DiplomaInit(u8);
+void Task_DiplomaReturnToOverworld(u8);
+
+extern const struct BgTemplate gUnknown_8415A08[2];
+extern const struct WindowTemplate gUnknown_8415A10[];
+extern const u8 gUnknown_8402650[];
+extern const u8 gUnknown_841B60E[];
+extern const u8 gUnknown_841B684[];
+extern u8 gUnknown_84147C0[];
+extern u8 gUnknown_8415954[];
+extern u8 gUnknown_841B698[];
+extern u8 gUnknown_841B68F[];
+extern u8 gUnknown_841B619[];
+extern u8 gUnknown_8415A04[];
+
+struct Diploma
+{
+    u8 state;
+    u8 gfxStep;
+    u8 callbackStep;
+	u8 padding;
+    u8 tilemapBuffer[0x1000];
```

Comment:
Do u16 tilemapBuffer[0x800]; instead, now you can remove the padding field and it will still work.